src/is_intersection_empty.jl

```diff
@@ -1365,3 +1365,78 @@ function is_intersection_empty(X::LazySet{N},
                                   {N<:Real}
     return is_intersection_empty(C, X, witness)
 end
+
+"""
+    is_intersection_empty(X::CartesianProductArray{N, S},
+                          Y::HPolyhedron{N}) where {N<:Real, S<:LazySet{N}}
+
+Check whether a Cartesian product array intersects with a H-polyhedron.
+
+### Input
+
+- `X` -- Cartesian product array of convex sets
+- `Y` -- H-polyhedron
+
+### Output
+
+`true` iff ``X ∩ Y = ∅ ``.
+"""
+function is_intersection_empty(X::CartesianProductArray{N, S},
```

Comment:
This method is *very* similar to `overapproximate`. Can you outsource the common parts (say, `intersection_helper`)?